src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy

```diff
@@ -164,6 +164,25 @@ class AutoRerouteSpec extends HealthCheckSpecification {
         pathHelper.getInvolvedIsls(pathAfterReroute1).each {
             assert northbound.getLink(it).availableBandwidth == flow.maximumBandwidth - 1 }
 
+        when: "Trigger auto reroute by blinking not involved(in flow path) isl"
+        def islToBlink = topology.islsForActiveSwitches.find {
+            [it.srcSwitch.dpId, it.dstSwitch.dpId].intersect([flow.source.switchId, flow.destination.switchId]).empty
+        }
+        antiflap.portDown(islToBlink.srcSwitch.dpId, islToBlink.srcPort)
+        def islToBlinkIsDown = true
+        antiflap.portUp(islToBlink.srcSwitch.dpId, islToBlink.srcPort)
+        islToBlinkIsDown = false
+
+        then: "System tries to reroute the DEGRADED flow"
+        and: "Flow remains DEGRADED and on the same path"
+        wait(rerouteDelay + WAIT_OFFSET) { //2 (after manual reroute) + 2 more reroute attempts (auto reroute + retry)
+            assert northbound.getFlowHistory(flow.flowId).size() == history.size() + 2 + 2
```

Comment:
This is not parallel-safe. There could be more reroutes triggered by other islands that will affect your history events counts

Reply:
fixed for manual/auto reroutes.